tests/integration/test_heartbeats.py

```diff
@@ -110,52 +94,57 @@ def verify_user_agent(output, expected, envname=None, marker=None):
 urls = [u for c in context.channels for u in Channel(c).urls()]
 urls.extend(u.rstrip("/") for u in context.channel_alias.urls())
 if any(".anaconda.cloud" in u for u in urls):
-    hb_url = "https://repo.anaconda.cloud/"
+    exp_host = "repo.anaconda.cloud:443"
 elif any(".anaconda.com" in u for u in urls):
-    hb_url = "https://repo.anaconda.com/"
+    exp_host = "repo.anaconda.com:443"
 elif any(".anaconda.org" in u for u in urls):
-    hb_url = "https://conda.anaconda.org/"
+    exp_host = "conda.anaconda.org:443"
 else:
-    hb_url = None
-if hb_url:
-    hb_url += "pkgs/main/noarch/activate-0.0.0-0.conda"
-print("Expected heartbeat url:", hb_url)
-print("Expected user agent tokens:", ",".join(expected))
+    raise RuntimeError("No heartbeat URL available.")
+exp_path = "/pkgs/main/noarch/activate-0.0.0-0.conda"
+print("Expected host:", exp_host)
+print("Expected path:", exp_path)
+print("Expected tokens:", ",".join(expected))
 need_header = True
-for hval in ("true", "false"):
-    os.environ["CONDA_ANACONDA_HEARTBEAT"] = hval
+for hval in ("true", "false", "delay"):
```

Comment:
"delay" mode is enabled, but the network connection is blocked, so we're testing that no heartbeat was actually sent.